v3/src/components/map/components/map-point-layer.tsx

```diff
@@ -123,6 +128,87 @@ export const MapPointLayer = observer(function MapPointLayer({mapLayerModel, set
     }
   }, [mapLayerModel.layerIndex, setPixiPointsLayer])
 
+  // TODO: Deleted attributes should be removed from the DataConfiguration, in
+  // which case the additional validation via the DataSet would be unnecessary.
+  const legendAttributeId = dataConfiguration.attributeID('legend')
+  const legendAttribute = dataset?.getAttribute(legendAttributeId)
+  const getLegendColor = legendAttribute ? dataConfiguration?.getLegendColorForCase : undefined
+  const lookupLegendColor = (aCaseData: CaseData) => {
+    return dataConfiguration.getLegendColorForCase(aCaseData.caseID) || pointDescription.pointColor
+  }
+
+  const refreshHeatmap = useDebouncedCallback(() => {
+    if (!heatmapCanvasRef.current || !dataset) return
+
+    // Initialize the simpleheat instance if necessary
+    if (!simpleheatRef.current) {
+      simpleheatRef.current = simpleheat(heatmapCanvasRef.current)
+      // Add shutterbug support. See: shutterbug-support.ts.
+      heatmapCanvasRef.current.classList.add("canvas-3d")
+    }
+
+    // Reset simpleheat
+    simpleheatRef.current.clear()
+    simpleheatRef.current.resize()
+
+    // If we should draw the heatmap, update simpleheat
+    const { isVisible, pointsAreVisible, pointType } = mapLayerModel
+    if (legendAttributeId && pointType === "heatmap" && pointsAreVisible && isVisible) {
+      // For some reason, the simpleheat canvas always changes to 300x150, so we have to scale the positions accordingly
+      const mapContainer = leafletMap.getContainer()
+      const mapRect = mapContainer.getBoundingClientRect()
+      const scaleX = 300 / mapRect.width
+      const scaleY = 150 / mapRect.height
```

Comment:
This is sufficiently bizarre (among other reasons) that I'd be inclined to consider other libraries, but given the speculative nature of the feature we can table that discussion for another day.